packages/material-ui-private-theming/src/ThemeProvider/ThemeProvider.test.js

```diff
@@ -55,9 +55,13 @@ describe('ThemeProvider', () => {
     const text = () => ref.current.textContent;
     function Test() {
       const theme = useTheme();
+      const themeRef = React.useRef(); // needed due to double-invokes effect in StrictMode
       React.useEffect(() => {
-        themes.push(theme);
-      });
+        if (themeRef.current !== theme) {
+          themes.push(theme);
```

Comment:
We're using this just to count renders. I added a `RenderCounter` utility to do that.

Reply:
fixed